cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -0,0 +1,514 @@
+							generators, NewGenConversion("conversion_generated", path))
+						return generators
+					},
+					FilterFunc: func(c *generator.Context, t *types.Type) bool {
+						return t.Name.Package == path
+					},
+				})
+		}
+	}
+	return packages
+}
+
+func findMember(t *types.Type, name string) (types.Member, bool) {
+	if t.Kind != types.Struct {
+		return types.Member{}, false
+	}
+	for _, member := range t.Members {
+		if member.Name == name {
+			return member, true
+		}
+	}
+	return types.Member{}, false
+}
+
+func isConvertible(in, out *types.Type) bool {
+	// FIXME: Check manually-written conversion functions.
+
+	// If one of the types is Alias, resolve it.
+	if in.Kind == types.Alias {
+		return isConvertible(in.Underlying, out)
+	}
+	if out.Kind == types.Alias {
+		return isConvertible(in, out.Underlying)
+	}
+
+	if in.Kind != out.Kind {
+		return false
+	}
+	switch in.Kind {
+	case types.Builtin, types.Struct, types.Map, types.Slice, types.Pointer:
+	default:
+		// We don't support conversion of other types yet.
+		return false
+	}
+	switch out.Kind {
+	case types.Builtin, types.Struct, types.Map, types.Slice, types.Pointer:
+	default:
+		// We don't support conversion of other types yet.
+		return false
+	}
+
+	switch in.Kind {
+	case types.Builtin:
+		// FIXME: Enough to be convertible - see AWSElastic
+		return in.Name == out.Name
+	case types.Struct:
+		convertible := true
+		for _, inMember := range in.Members {
+			// Check if there is an out member with that name.
+			outMember, found := findMember(out, inMember.Name)
+			if !found {
+				return false
+			}
+			convertible = convertible && isConvertible(inMember.Type, outMember.Type)
+		}
+		return convertible
+	case types.Map:
+		return isConvertible(in.Key, out.Key) && isConvertible(in.Elem, out.Elem)
+	case types.Slice:
+		return isConvertible(in.Elem, out.Elem)
+	case types.Pointer:
+		return isConvertible(in.Elem, out.Elem)
+	}
+	glog.Fatalf("All other types should be filtered before")
+	return false
+}
+
+const (
+	apiPackagePath        = "k8s.io/kubernetes/pkg/api"
+	conversionPackagePath = "k8s.io/kubernetes/pkg/conversion"
+)
+
+// genConversion produces a file with a autogenerated conversions.
+type genConversion struct {
+	generator.DefaultGen
+	targetPackage string
+	imports       namer.ImportTracker
+	typesForInit  []*types.Type
+}
+
+func NewGenConversion(sanitizedName, targetPackage string) generator.Generator {
+	return &genConversion{
+		DefaultGen: generator.DefaultGen{
+			OptionalName: sanitizedName,
+		},
+		targetPackage: targetPackage,
+		imports:       generator.NewImportTracker(),
+		typesForInit:  make([]*types.Type, 0),
+	}
+}
+
+func (g *genConversion) Namers(c *generator.Context) namer.NameSystems {
+	// Have the raw namer for this file track what it imports.
+	return namer.NameSystems{"raw": namer.NewRawNamer(g.targetPackage, g.imports)}
+}
+
+func (g *genConversion) convertibleWithinPackage(inType, outType *types.Type) bool {
```

Comment:
I'm not convinced, but I'm not good at naming. Will change (and can change it again in the future if needed).